addons/account/report/account_invoice_report.py

```diff
@@ -95,7 +104,14 @@ def _select(self):
                 line.quantity / NULLIF(COALESCE(uom_line.factor, 1) / COALESCE(uom_template.factor, 1), 0.0) * (CASE WHEN move.move_type IN ('in_invoice','out_refund','in_receipt') THEN -1 ELSE 1 END)
                                                                             AS quantity,
                 -line.balance * currency_table.rate                         AS price_subtotal,
-                line.price_total * (CASE WHEN move.move_type IN ('in_invoice','out_refund','in_receipt') THEN -1 ELSE 1 END)
+                -- line.price_total needs to be converted to line.company_currency_id (using the rate at the time of the invoice),
+                -- get the correct sign and be converted to the currency of the active company (using the rate from today)
+                ROUND(line.price_total
+                          * (CASE WHEN move.move_type IN ('in_invoice','out_refund','in_receipt') THEN -1 ELSE 1 END)
+                          * COALESCE(company_currency_conversion.rate, 1.0)
+                          / COALESCE(line_currency_conversion.rate, 1.0)
+                      , company_currency.decimal_places)
+                    * currency_table.rate
```

Comment:
Wouldn't something like have the same result at a much lower cost?
```suggestion
             -line.balance * currency_table.rate * (line.price_total / line.price_subtotal)
                                                 AS price_subtotal,
```

Reply:
Might lead to some rounding differences that could accumulate though